src/CodeGen/ODataT4CodeGenerator.cs

```diff
@@ -425,6 +438,124 @@ public void ValidateAndSetIgnoreUnexpectedElementsAndAttributesFromString(string
     this.IgnoreUnexpectedElementsAndAttributes = boolValue;
 }
 
+/// <summary>
+/// Set the CustomHttpHeaders property from the custom http headers string.
+/// </summary>
+/// <param name="stringValue">The value to set.</param>
+public void ValidateAndSetCustomHttpHeadersFromString(string stringValue) 
+{
+    if (String.IsNullOrWhiteSpace(stringValue)) {
+        return;
+    }
+
+    this.CustomHttpHeaders = new Dictionary<string, string>();
+
+    if (ValidateParantheses(stringValue) && ValidateFirstLast(stringValue) && ValidateDelimeters(stringValue)) 
+    {
+        string header = stringValue.TrimEnd('}', ' ');
+        string[] elements = header.Split('}');
+        foreach (var element in elements)
+        {
+            string _element = element.TrimStart(',', '{', ' ');
+            string[] array = _element.Split(':');
+            CustomHttpHeaders.Add(array[0], array[1]);
+        }
+    }
+    else 
+    {
+        throw new ArgumentException("Custom http headers string format should be '{headerName:value},{headerName:value}'");
+    }
+
+}
+
+private static bool ValidateParantheses(string _input)
+{
+
+    Stack<char> stack = new Stack<char>();
+
+    foreach (char c in _input)
+    {
+        if(stack.Count > 1)
+        {
+            return false;
+        }
+        if (c == '{')
+        {
+            stack.Push(c);
+        }                    
+        else if(c == '}')
+        {
+            if(stack.Count > 0 && stack.Peek() == '{')
+            {
+                stack.Pop();
+            }
+            else
+            {
+                return false;
+            }
+        }
+    }
+    return stack.Count() == 0 ? true : false;

```

Comment:
>stack.Count() == 0 ? true : false; [](start = 11, length = 34)

If the parentheses can't match, can you throw exception directly here?

Reply:
This was fixed after I changed the input to be a verbatim string